firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/impl/TestUtils.java

```diff
@@ -101,27 +98,9 @@ static void awaitAsyncOperations(ExecutorService executorService) throws Interru
     shadowOf(getMainLooper()).idle();
   }
 
-  /** Await a specified number of progress events on an {@link UpdateTask}. */
-  static List<UpdateProgress> awaitProgressEvents(UpdateTask updateTask, int count)
+  /** Await a specified number of progress events being added to the given collection. */
+  static void awaitProgressEvents(Collection<UpdateProgress> progressEvents, int count)
```

Comment:
Sorry, but I keep having ideas for this:

I think generalizing this would actually make it more readable. It doesn't need to be full-featured like http://google3/java/com/google/testing/util/WaitForCondition.java

How about:

```
  static void waitForCondition(BooleanSupplier condition) {
      ExecutorService executor = TestOnlyExecutors.blocking();
    executor.execute(
        () -> {
          while (!condition.getAsBoolean()) {
            try {
              Thread.sleep(50);
            } catch (InterruptedException e) {
              throw new RuntimeException("Interrupted while waiting for progress events", e);
            }
          }
        });
    executor.awaitTermination(500, TimeUnit.MILLISECONDS);
    assertThat(condition.getAsBoolean());
  }
```

That way the caller becomes more expressive:

```
  waitForCondition(() -> progressEvents.size() == 3);
```

Reply:
No I think this makes more sense now that most of the progress event specific logic is gone. I also removed the Executor since I realized that wasn't actually adding value. It was also inadvertently causing us to wait a minimum of 500ms each time.